addons/l10n_ch/models/account_invoice.py

```diff
@@ -72,6 +70,12 @@ def _get_invoice_reference_ch_partner(self):
         self.ensure_one()
         return self.get_l10n_ch_qrr_number()
 
+    def _is_swiss_qr_iban(self):
+        """QR code is also printed if the fiscal country is not Switzerland but the receivale account is eligible"""
+        self.ensure_one()
+        iban = (self.partner_bank_id.acc_number or '').replace(' ', '')
+        return self.partner_bank_id.acc_type == 'iban' and iban.startswith('CH') and iban[4:9].isdigit() and 30000 <= int(iban[4:9]) <= 31999
```

Comment:
and currency = CHF ?

Reply:
in this method we are only checking that it is a swiss qr_iban

Reply:
As of now, we will try to generate the qr even if:
- my company is not swiss (this case -> french)
- **the currency of the move is not ch**
- the iban is a swiss qr iban

see video on runbot saas-18.2
[Screencast from 27-03-25 11:36:57.webm](https://github.com/user-attachments/assets/6f62bfaf-ffbf-4fae-a8f5-270afc96e679)